torch/csrc/distributed/c10d/comm.h

```diff
@@ -38,7 +38,11 @@ class GradBucket {
 // DDP's c10d reducer allows communication hooks defined as a sub class
 // of CommHookInterface. CommHookInterface is an abstract class and can
 // be used to implement both Python and CPP hooks.
+#ifdef _WIN32
+struct CommHookInterface {
+#else
 struct TORCH_API CommHookInterface {
+#endif
```

Comment:
Sounds like `C10_EXPORT` macro would solve everyones problems here, wouldn't it? (Although in order to support DDP hooks on Windows we perhaps need to add something like TORCH_PYTHON_API)
```suggestion
struct C10_EXPORT CommHookInterface {
```